smart_open/smart_open_lib.py

```diff
@@ -28,11 +28,23 @@
 import requests
 import io
 
+from functools import partial
+
 
 IS_PY2 = (sys.version_info[0] == 2)
 
 if IS_PY2:
     import httplib
+    if sys.version_info[1] == 6:
+        import copy_reg
```

Comment:
@shaform What is this cleverness?

We try not to write code that is too clever, and if we must, we comment it heavily.

Please include detailed comments about the motivation and behaviour of these blocks of code.